build-logic/src/main/kotlin/org/terasology/gradology/module_build.kt

```diff
@@ -48,9 +48,11 @@ class ModuleMetadataForGradle(private val moduleConfig: ModuleMetadata) {
     val group: String = TERASOLOGY_MODULES_GROUP
 
     fun engineVersion(): String {
-        return moduleConfig.dependencies.filterNotNull()
+        return moduleConfig.dependencies
+            .filterNotNull()
             .find { it.id.toString() == ENGINE_MODULE_NAME }
-            ?.versionRange()?.toString() ?: "+"
+            ?.let(this::versionStringFromGestaltDependency)
+            ?: "+"
```

Comment:
thought: genuinely not sure what to think about this usage of `?.let()`.

I know this sort of construct is one approach to these situations where we're doing chains of "only if not null" operations, but it's an area of Kotlin I still feel much less than fluent in.

The IDE tells me that we could leave off the explicit `this`:
```kotlin
return allThatStuff
    ?.let(::versionStringFromGestaltDependency)
    ?: "+"
```

or expand it back out to the block form:
```kotlin
return allThatStuff
    ?.let { versionStringFromGestaltDependency(it) }
    ?: "+"
```

I find myself preferring the block form to read, but I can definitely also appreciate the functional perspective of "this is calling exactly this function, we don't need to introduce another scope for it!"

Reply:
I was actually using VS Code for this one - maybe IntelliJ support would have helped me here. 

Personally, the block form is _less_ readable for me :thinking: but I also don't understand whether `let` is the right thing to use here. There are other words to use, I think, such as `apply`, which might fit better...

Actually, I was looking for `map` as it would be called in Scala...